src/site/content/en/blog/text-fragments/index.md

```diff
@@ -0,0 +1,422 @@
+to create an extension for browsers that do not support Text Fragments natively
+where the functionality is implemented in JavaScript.
+
+### Security
+
+Text fragment directives are invoked only on full (non-same-page) navigations
+that are the result of a
+[user activation](https://html.spec.whatwg.org/multipage/interaction.html#tracking-user-activation).
+Additionally, navigations originating from a different origin than the destination
+will require the navigation to take place in a
+[`noopener`](https://html.spec.whatwg.org/multipage/links.html#link-type-noopener) context,
+such that the destination page is known to be sufficiently isolated.
+Text fragment directives are only applied to the main frame.
+This means that text will not be searched inside iframes,
+and iframe navigation will not invoke a text fragment.
+
+### Privacy
+
+It is important that implementations of the Text Fragments specification
+do not leak whether a text fragment was found on a page or not.
+While element fragments are fully under the control of the original page author
+(remember how there was no way to link to the section
+I wanted to link to before in my running example),
+text fragments can be created by anyone.
+
+Imagine I ran an evil ad network `evil-ads.example.com`.
+Further imagine that in one of my ad iframes I opened a hidden cross-origin iframe
+to `dating.example.com` with a Text Fragment URL
+<code>dating.example.com<strong>#:~:text=Log%20Out</strong></code>
+when the user interacts with the ad.
+If the text "Log Out" is found, I know the victim is currently logged in to `dating.example.com`,
+which I could use for user profiling.
+Since a naive Text Fragments implementation might decide
+that a successful match should cause a focus switch,
+on `evil-ads.example.com` I could listen for the `blur` event and thus know when a match occurred.
+In Chrome, we have implemented Text Fragments in such a way that the above scenario cannot happen.
+
+Another attack might be to exploit network traffic based on scroll position.
+Assume I had access to network traffic logs of my victim, like as the admin of a company intranet.
+Now imagine there existed a long human resources document "what to do if you suffer from…"
+and then a list of conditions like "burn out", "anxiety", etc.
+I could place a tracking pixel next to each item on the list.
+If I then determine that loading the document temporally co-occurs
+with the loading of the tracking pixel next to, say, the "burn out" item,
+I can then, as the intranet admin, determine that an employee has clicked through
+on a text fragment link with `:~:text=burn%20out`
+that the employee may have assumed was confidential and not visible to anyone.
+Since this example is somewhat contrived
+and its exploitation requires very specific preconditions to be met,
+on the Chrome team we have decided to implement scroll on navigation,
+which is one of the preconditions.
```

Comment:
I do't understand why scroll navigation makes text fragments safer.

Reply:
The opposite. We believe the risk scroll navigation adds is manageable.

Reply:
Your wording doesn't convey that.

Reply:
I have rephrased.